assets/javascripts/discourse/connectors/topic-footer-main-buttons-before-create/assign-button.js.es6

```diff
@@ -3,7 +3,7 @@ import { ajax } from 'discourse/lib/ajax';
 
 export default {
   shouldRender(args, component) {
-    const needsButton = component.currentUser && component.currentUser.get('staff');
+    const needsButton = component.currentUser && component.currentUser.get('staff') && component.siteSettings.assign_enabled;
```

Comment:
Ah I know why! It needs the path to include `discourse-assign` which it does not with this configuration.

Try moving `assets/javascripts/discourse` within the plugin to be `assets/javascripts/discourse-assign` - then the matcher will work.